### PR TITLE
chore: apply clippy lints to public API

### PR DESCRIPTION
**Pull request by thomaseizinger**
## Description

By default, clippy doesn't suggest changes if it would break the public API. Whilst a sensible default, it makes us miss certain opportunities to follow conventions correctly. When a new lint is added, we can always temporarily allow it and make a change in the next breaking release but I think it is better to be aware of the lint than having it automatically silenced.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: identity/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.2.6
+
+- Make `PeerId::to_bytes` and `PeerId::to_base58` take `self` by value to follow Rust convention of `Copy` types.
+  See [PR XXXX](https://github.com/libp2p/rust-libp2p/pull/XXXX).
```
Comment (thomaseizinger) [inline identity/CHANGELOG.md]:
```suggestion
  See [PR 4653](https://github.com/libp2p/rust-libp2p/pull/4653).
```

File: protocols/gossipsub/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,12 @@
 - Deprecate `gossipsub::Config::idle_timeout` in favor of `SwarmBuilder::idle_connection_timeout`.
   See [PR 4648].
 
+<!-- Interal changes:
+
+- Allow new clippy lint.
+
+-->
```
Comment (thomaseizinger) [inline protocols/gossipsub/CHANGELOG.md:10]: My new changelog lint forces us to always update the changelog. I'd suggest we just make an HTML comment in the changelogs for internal changes.

Thoughts?

File: identity/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 0.2.6
+
+- Make `PeerId::to_bytes` and `PeerId::to_base58` take `self` by value to follow Rust convention of `Copy` types.
+  See [PR 4653](https://github.com/libp2p/rust-libp2p/pull/4653).
```
Comment (mxinden) [inline identity/CHANGELOG.md:4]: Isn't this a breaking change?

File: protocols/gossipsub/CHANGELOG.md (hunk 2 above)
Comment (mxinden) [reply protocols/gossipsub/CHANGELOG.md:10]: Works for me.

File: identity/CHANGELOG.md (hunk 3 above)
Comment (mxinden) [reply identity/CHANGELOG.md:4]: Ah, never mind. Given that it is a `Copy` type, `&self` -> `self` is fine.

**Review by mxinden — APPROVED** [review]

**Pull request merged**